libr/include/r_asm.h

```diff
@@ -66,21 +66,25 @@ enum {
 };
 
 typedef struct r_asm_op_t {
-	int size; // instruction size
-	int bitsize; // instruction size in bits (or 0 if fits in 8bit bytes)
+	int size; // instruction size (must be deprecated. just use buf.len
+	int bitsize; // instruction size in bits (or 0 if fits in 8bit bytes) // wtf why dupe this field? :D
 	int payload; // size of payload (opsize = (size-payload))
 	// But this is pretty slow..so maybe we should add some accessors
-	ut8  buf[R_ASM_BUFSIZE + 1];
-	char buf_asm[R_ASM_BUFSIZE + 1];
-	char buf_hex[R_ASM_BUFSIZE + 1];
-	RBuffer *buf_inc;
+	RStrBuf buf;
+	RStrBuf buf_asm;
+	RStrBuf buf_hex;
+	RBuffer *buf_inc; // must die
 } RAsmOp;
 
 typedef struct r_asm_code_t {
+#if 1
 	int len;
 	ut8 *buf;
 	char *buf_hex;
 	char *buf_asm;
+#else
```

Comment:
Please either replace those fields with rasmop or remove the else branch. Having if 1 everywhere is useless

Reply:
its a reminder to do in my next refactor. rasmcode needs to be changed too